utils/fsgdf.cpp

```diff
@@ -880,28 +884,56 @@ int gdfCountItemsInString(char *str) {
   is -1, then it returns the last item. item is a string that
   must be freed by the caller.
   ------------------------------------------------------------------*/
-char *gdfGetNthItemFromString(char *str, int nth) {
+char *gdfGetNthItemFromString(const char *str, const int nth) {
   char *item;
-  int nitems,n;
-  static char fmt[2000], tmpstr[2000];
-
-  memset(fmt,'\0',2000);
-  memset(tmpstr,'\0',2000);
+  int nitems;
 
   nitems = gdfCountItemsInString(str);
-  if (nth < 0) nth = nitems-1;
   if (nth >= nitems) {
     printf("ERROR: asking for item %d, only %d items in string\n",nth,nitems);
     printf("%s\n",str);
     return(NULL);
   }
 
-  for (n=0; n < nth; n++) sprintf(fmt,"%s %%*s",fmt);
-  sprintf(fmt,"%s %%s",fmt);
-  //printf("fmt %s\n",fmt);
-  sscanf(str,fmt,tmpstr);
+  const std::string src(str);
+  std::vector<std::string> items;
+  std::string tmpstr;
+  bool inItem = !std::isspace(src.at(0));
+  for(auto it=src.begin(); it!=src.end(); ++it ) {
```

Comment:
The first time I tried, this loop definitely had some horrible bugs.... yet all the tests still passed. This concerns me. It might still contain some unpleasant bugs.

Reply:
okay thanks, I'll run some extra tests on this section